plugin/inc/ticket.class.php

```diff
@@ -1222,4 +1215,25 @@ public function updateConsumable($ticketId, $post): void
 
     }
 
+    public function updateEffectiveDate($ticketId, $post): void
+    {
+        // If ticket status is Ticket::SOLVED or Ticket::CLOSED, effective date should not change.
+        // We presume that in such cases effective date is always set.
+        if ($this->fields['status'] == Ticket::SOLVED && $this->fields['status'] == Ticket::CLOSED) {
```

Comment:
1. The code does not match the comment
2. Always use `===` when possible, see [this article](https://marketsplash.com/tutorials/php/php-comparison/) for more details.
```suggestion
        if ($this->fields['status'] === Ticket::SOLVED || $this->fields['status'] === Ticket::CLOSED) {
```